### ISSUE TITLE
Django 1.9 compatibility

### ISSUE THREAD

**Issue by rfleschenberg** [makinacorpus/django-safedelete#34]
``models.py`` calls ``safedelete_mixin_factory(SOFT_DELETE)`` at import time. This breaks against the current Django master branch, so it will likely be broken on Django 1.9.

I don't fully understand the issue yet. Can we move that call somewhere else, so it is executed after the app registry is ready?

Here is an example traceback:
```
Traceback (most recent call last):
  File "./manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/rene/.virtualenvs/academy/lib/python3.4/site-packages/django/core/management/__init__.py", line 331, in execute_from_command_line
    utility.execute()
  File "/home/rene/.virtualenvs/academy/lib/python3.4/site-packages/django/core/management/__init__.py", line 305, in execute
    django.setup()
  File "/home/rene/.virtualenvs/academy/lib/python3.4/site-packages/django/__init__.py", line 18, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/rene/.virtualenvs/academy/lib/python3.4/site-packages/django/apps/registry.py", line 85, in populate
    app_config = AppConfig.create(entry)
  File "/home/rene/.virtualenvs/academy/lib/python3.4/site-packages/django/apps/config.py", line 86, in create
    module = import_module(entry)
  File "/home/rene/.virtualenvs/academy/lib64/python3.4/importlib/__init__.py", line 109, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
  File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
  File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1129, in _exec
  File "<frozen importlib._bootstrap>", line 1471, in exec_module
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "/home/rene/.virtualenvs/academy/lib/python3.4/site-packages/safedelete/__init__.py", line 5, in <module>
    from .models import safedelete_mixin_factory
  File "/home/rene/.virtualenvs/academy/lib/python3.4/site-packages/safedelete/models.py", line 127, in <module>
    SoftDeleteMixin = safedelete_mixin_factory(SOFT_DELETE)
  File "/home/rene/.virtualenvs/academy/lib/python3.4/site-packages/safedelete/models.py", line 37, in safedelete_mixin_factory
    class Model(models.Model):
  File "/home/rene/.virtualenvs/academy/lib/python3.4/site-packages/django/db/models/base.py", line 94, in __new__
    app_config = apps.get_containing_app_config(module)
  File "/home/rene/.virtualenvs/academy/lib/python3.4/site-packages/django/apps/registry.py", line 239, in get_containing_app_config
    self.check_apps_ready()
  File "/home/rene/.virtualenvs/academy/lib/python3.4/site-packages/django/apps/registry.py", line 124, in check_apps_ready
    raise AppRegistryNotReady("Apps aren't loaded yet.")
django.core.exceptions.AppRegistryNotReady: Apps aren't loaded yet.
```

**Comment by Gagaro** [comment]
Thanks for the report. I wonder if it has something to do with the fact that we import everything in `__init__.py`.

**Comment by rfleschenberg** [comment]
As far as I understand, you cannot create model classes in ``__init__.py``, because ``__init__.py`` runs before the app registry is ready (the app registry has to import the app). Removing the import from ``__init__.py`` fixes the issue.
I just did not do this in the PR because I wanted to break the API as little as possible, and I thought moving ``SoftDeleteMixin`` would have less impact. But I actually think it is best to not import anything at all in ``__init__.py``. Maybe create an ``api.py`` module if you want to be able to import all the public objects from one place.

**Comment by Gagaro** [comment]
I also think there shouldn't be any import in `__init__.py`. I looked at some other projects to see how they handle that but I don't think they are compatible with the 1.9 yet. Maybe we should wait to see how others project do that or discuss it on the mailings list to have other inputs.

**Comment by Gagaro** [comment]
Solved by #36
